app/src/main/java/fr/free/nrw/commons/upload/UploadService.java

```diff
@@ -105,7 +106,7 @@ public void onProgress(long transferred, long total) {
             notificationManager.notify(notificationTag, NOTIFICATION_UPLOAD_IN_PROGRESS, curNotification.build());
 
             contribution.setTransferred(transferred);
-            contributionDao.save(contribution);
+            ExecutorUtils.get().submit(() -> appDatabase.getContributionDao().save(contribution));
```

Comment:
Not using `Single`s here?

Reply:
Updated to Singles, thanks for reminding :)